ml-agents/mlagents/trainers/torch/components/reward_providers/gail_reward_provider.py

```diff
@@ -69,31 +69,23 @@ def __init__(self, specs: BehaviorSpec, settings: GAILSettings) -> None:
         self._use_vail = settings.use_vail
         self._settings = settings
 
-        state_encoder_settings = NetworkSettings(
+        encoder_settings = NetworkSettings(
             normalize=False,
             hidden_units=settings.encoding_size,
             num_layers=2,
             vis_encode_type=EncoderType.SIMPLE,
             memory=None,
         )
-        self._state_encoder = NetworkBody(
-            specs.observation_shapes, state_encoder_settings
-        )
-
         self._action_flattener = ModelUtils.ActionFlattener(specs)
 
-        encoder_input_size = settings.encoding_size
         if settings.use_actions:
-            encoder_input_size += (
-                self._action_flattener.flattened_size + 1
+            self.encoder = NetworkBody(
+                specs.observation_shapes,
+                encoder_settings,
+                self._action_flattener.flattened_size + 1,
             )  # + 1 is for done
-
-        self.encoder = torch.nn.Sequential(
-            linear_layer(encoder_input_size, settings.encoding_size),
-            Swish(),
-            linear_layer(settings.encoding_size, settings.encoding_size),
-            Swish(),
-        )
+        else:
+            self.encoder = NetworkBody(specs.observation_shapes, encoder_settings)
```

Comment:
Why aren't there dones here?

Reply:
Also: can we do 
```
unencoded_size = (self._action_flattener.flattened_size if settings.use_actions else 0) + 1 # dones
self.encoder = NetworkBody(specs.observation_shapes, encoder_settings, unencoded_size)
```
Of course you can move the +1 for the dones into the parenthesis if statement as well

Reply:
We only put the done if the actions are also there. The setting `use_action` is in fact `use_action_and_also_done`

Reply:
We only use `dones` when the action is included in tf also. Additionally, updated the code to reflect @ervteng's suggestion